### PR TITLE
Remove most cyclic dependencies in front-end

### PR DESCRIPTION
**Pull request by philippotto**
This PR removes most, but not all cyclic dependencies. I'd suggest to tackle the remaining one's once we've got some air for that. I added a check to the CI, though, that ensures that a new PR won't increase the amount of cycles. I hope this works alright (if not we can disable that check again).

Changed code is 99% just moved around. The singletons module is new and due to it's explicit set-functions breaks up some critical cyclic dependencies (e.g., Model, Store etc).

### URL of deployed dev instance (used for testing):
- https://cyclicdeps.webknossos.xyz

### Steps to test:
- CI

### Issues:
- contributes to #3401

------
(Please delete unneeded items, merge only when none are left open)
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/model/accessors/flycam_accessor.ts [hunk 1]
```diff
@@ -307,7 +312,7 @@ export function getValidTaskZoomRange(
   state: OxalisState,
   respectRestriction: boolean = false,
 ): [number, number] {
-  const defaultRange = [userSettings.zoom.minimum, Infinity];
+  const defaultRange = [baseDatasetViewConfiguration.zoom.minimum, Infinity];
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/model/accessors/flycam_accessor.ts:315]: Do you know why typescript did not complain about this, before?

**Review by daniel-wer — COMMENTED** [review]
Splendid work, thank you for taking the time! I'll do some quick monkey testing and will report back, afterwards.

File: frontend/javascripts/oxalis/model/accessors/flycam_accessor.ts (hunk 1 above)
Comment (philippotto) [reply frontend/javascripts/oxalis/model/accessors/flycam_accessor.ts:315]: If I remember correctly, both code versions are fine value-wise (i.e., they are set to the same value), but the old one caused a cyclic dependencies, about which TS doesn't care :)

Comment (daniel-wer) [reply frontend/javascripts/oxalis/model/accessors/flycam_accessor.ts:315]: Ah, I searched for `zoom` in the `user_settings.schema` and couldn't find it, but there's a `...baseDatasetViewConfiguration,` in there that I missed :+1:

**Review by daniel-wer — APPROVED** [review]
LGTM :tada:

**Pull request merged**